### ISSUE TITLE
Decide which versioning scheme to use

### ISSUE THREAD

**Issue by phaer** [kube-hetzner/kube-hetzner#80]
Recently, the master branch has been quite unstable and required users to either stay on an unsupported, older commit of kube-hetzner or to re-provision their whole cluster. 

I believe that the time would be right to agree on a versioning scheme and implement at least a minimum of a release process to communicate breaking changes more clearly.

**Comment by mysticaltech** [comment]
@phaer You are right and have been feeling bad about that too lately. Have realized too how users are confused. Thanks for taking this initiative. Definitly, semver is the way to go IMHO. Also, thanks for the project, that will get us more organized too :)
